### ISSUE TITLE
bluetooth low energy devices stopped working on newer linux kernels (6.4) 

### ISSUE THREAD

**Issue by MangoIV** [NixOS/nixpkgs#242313]
### Describe the bug

For a while not I cannot pair any bluetooth keyboards anymore, the keyboards have been confirmed to work fine by connecting to another central device. 

### Steps To Reproduce

- nixos version is 23.05
- try to connect via `bluetoothctl`
  - `scan on`
  - `scan off` 
  - `agent on` 
  - `default-agent`
  - `connect <MAC address>`
  - tries to connect and pair but then immediately aborts
 
### Expected behavior

the keyboard connects and pairs fine 

### Screenshots

![image](https://github.com/NixOS/nixpkgs/assets/40720523/3b0b3917-4bcf-45c3-a389-ca83bc1f72dc)

### Additional context

- everything to do with bluetooth has been incredibly unstable, stuff randomly stops working, there are hard to reproduce bugs and failures all over the place, the tooling is not well documented and so on. I know this is not a NixOS issue per se but it would be nice if we could do something about it. If anybody had any input on how other Linux distribution deal with bluetooth stacks to be stable and nice to use, that would be really awesome. 

- My entire configuration can be found at https://git.sr.ht/~mangoiv/dotfiles/tree/dad63d0984033c7a9e55b107848359420353ce37

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.4.1-zen1, NixOS, 23.11 (Tapir), 23.11.20230706.3c74875`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix Super) 2.14.0pre20230128_792d83f`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by kelszo** [comment]
I had a similar problem where bluetooth LE stopped working and I couldn't find my keyboard. I was also running kernel version 6.4. Downgrading the kernel to 6.1 solved it for me.

**Comment by MangoIV** [comment]
Thank you for that information, do you know what specifically causes this behaviour?

**Comment by kelszo** [comment]
Sadly not. I couldn't find any good error logs/messages and couldn't find anything relevant on the kernels bugzilla. What I can say is that this problem was replicated on two different devices (HP Elitebook and Dell Precision) both running NixOS on the latest kernel, downgrading to 6.1 helped on both.

Also other bluetooth devices worked, seems to be isolated to just LE devices. So most likely some bug in Bluez kernel modules for LE?

Did downgrading help you?

**Comment by MangoIV** [comment]
No, though I haven’t tried yet - should do that - I mean it’s not a long term solution but it might resolve the issue for now.

**Comment by kelszo** [comment]
Most likely related to https://bugzilla.kernel.org/show_bug.cgi?id=217651

Seems to be a patch in the pipeline so let's hope it is related so we can return to the latest kernel version :)

**Comment by kelszo** [comment]
It now works for me with the latest kernel version 6.4.9 (nixpkgs rev `ce5e4a6ef2e59d89a971bc434ca8ca222b9c7f5e`)

**Comment by MangoIV** [comment]
Nice thank you for the heads up!

**Comment by pearsche** [comment]
I can't pair my bluetooth 5.0 mouse  anymore and that patch/thing doesn't seem like it helped ...

**Comment by MangoIV** [comment]
Yes for me too, had to downgrade to 6.1 again.

**Comment by pearsche** [comment]
I have tested opensuse's live cd iso and the issue is present there too. Someone has to report this on kernel.org

**Comment by MangoIV** [comment]
this is not present for all bluetooth LE devices apparently, I have a pointing device that is working just fine, whereas my keyboard with ZMK firmware doesn't 
I'm starting to think this might be an issue with the clients bluetooth stack, perhaps zephyros

**Comment by kelszo** [comment]
Can chime in with when I had this problem it was also with a keyboard running ZMK firmware

**Comment by MangoIV** [comment]
Alright. Maybe it's not a kernel problem after all

**Comment by MangoIV** [comment]
@kelszo there might be fix for you https://github.com/zmkfirmware/zmk/issues/1487#issuecomment-1836675169

**Comment by MangoIV** [comment]
I think it's pretty clear by now that this issue is caused by ZMK. If you run into this issue, please refer to the discussion above which should solve your problems.

**Comment by buttergrillcorn** [comment]
> @kelszo there might be fix for you [zmkfirmware/zmk#1487 (comment)](https://github.com/zmkfirmware/zmk/issues/1487#issuecomment-1836675169)

Thank you, finally found what's causing the issue after days of research.
I can confirm that after enabling the experimental features for ZMK does indeed fix the problem.
